src/main/java/org/apache/maven/plugins/dependency/tree/TreeMojo.java

```diff
@@ -309,6 +352,64 @@ public void setSkip( boolean skip )
 
     // private methods --------------------------------------------------------
 
+
+    private DependencyNode convertToCustomDependencyNode( org.eclipse.aether.graph.DependencyNode node )
+    {
+        DefaultDependencyNode rootNode = new DefaultDependencyNode( null,
+                convertAetherArtifactToMavenArtifact( node ), null, null, null );
+
+        rootNode.setChildren( new ArrayList<DependencyNode>() );
+
+        for ( org.eclipse.aether.graph.DependencyNode child : node.getChildren() )
+        {
+            rootNode.getChildren().add( buildTreeDfs( rootNode, child ) );
+        }
+
+        return rootNode;
+    }
+
+    private DependencyNode buildTreeDfs( DependencyNode parent, org.eclipse.aether.graph.DependencyNode child )
```

Comment:
buildTreeDepthFirst? why does it matter whether we build tree depth first or breadth first if it's the same tree?